### PR TITLE
#121 Feauture/Delete user and edit his role

### PR DESCRIPTION
**Pull request by deimonbody**
Delete user by admin:
-route "/api/users/{id}" - delete
Edit role by admin 
-route "/api/users/{id}" - put

### CODE REVIEW

File: frontend/src/components/admin/styles.scss [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+.users__row {
+  &:hover button {
+    visibility: visible !important;
+  }
+}
+.user-control-btn {
+  margin-right: 1rem;
+
+  @media screen and (max-width: 767px) {
```
Comment (nikita-remeslov) [inline frontend/src/components/admin/styles.scss]: You can use bootstrap breakpoints

File: backend/src/data/seeders/index.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { createConnection } from 'typeorm';
 
 import ormconfig from '~/config/ormconfig';
 
-import CompanySeeder from './company.seeder';
+// import CompanySeeder from './company.seeder';
```
Comment (nikita-remeslov) [inline backend/src/data/seeders/index.ts]: ?

File: backend/src/data/seeders/index.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ import SkillSeeder from './skills.seeder';
 
 async function Connection(): Promise<void> {
   await createConnection(ormconfig);
-  await CompanySeeder.execute();
+  // await CompanySeeder.execute();
```
Comment (nikita-remeslov) [inline backend/src/data/seeders/index.ts]: ?

**Review by nikita-remeslov — APPROVED** [review]

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Resolve comments

File: backend/src/services/user.service.ts [hunk 4]
```diff
@@ -250,3 +255,51 @@ export const updateUserAvatar = async (
...
+  });
+
+  if (!userInstance)
+    throw new HttpError({
+      status: HttpCode.NOT_FOUND,
+      message: 'User with this id does not exist',
+    });
+
+  userRoleInstance[0].remove();
+  userInstance.remove();
```
Comment (aleksandr-kuharenko) [inline backend/src/services/user.service.ts]:
```suggestion
  await userRoleInstance[0].remove();
  await userInstance.remove();
```

File: backend/src/services/user.service.ts [hunk 5]
```diff
@@ -250,3 +255,51 @@ export const updateUserAvatar = async (
...
+    },
+  });
+  if (!userRoleInstance) {
+    throw new HttpError({
+      status: HttpCode.NOT_FOUND,
+      message: 'Can`t change Role for this role',
+    });
+  }
+  userRoleInstance[0].role = roleType;
+  userRoleInstance[0].save();
```
Comment (aleksandr-kuharenko) [inline backend/src/services/user.service.ts]:
```suggestion
  await userRoleInstance[0].save();
```

File: backend/src/services/user.service.ts [hunk 6]
```diff
@@ -250,3 +255,51 @@ export const updateUserAvatar = async (
...
+  userRoleInstance[0].remove();
+  userInstance.remove();
+
+  return { success: true, message: 'User deleted successfully' };
+};
+
+export const changeUserRole = async (
+  id: User['id'],
+  { roleType }: IChangeRoleProps,
+): Promise<any> => {
```
Comment (aleksandr-kuharenko) [inline backend/src/services/user.service.ts]: remove any

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Review by AndriyCh1 — APPROVED** [review]

**Pull request merged**